pkg/tfgen/docs.go

```diff
@@ -33,12 +33,23 @@ import (
 	"github.com/pulumi/pulumi-terraform-bridge/pkg/tfbridge"
 )
 
+// Either doc or arguments
+type argument struct {
+	// The description for this argument
+	description string
+	// The arguments for this argument
+	arguments map[string]string
+	// Whether this argument was derived from a nested object. Used to determine
+	// whether to append descriptions that have continued to the following line
+	isNested bool
+}
+
 // parsedDoc represents the data parsed from TF markdown documentation
 type parsedDoc struct {
 	// Description is the description of the resource
 	Description string
 	// Arguments includes the names and descriptions for each argument of the resource
-	Arguments map[string]string
+	Arguments map[string]*argument
```

Comment:
Just for my own understanding: This is now a map of "object name" to arguments associated with that object? And an empty string for the object name represents the top-level properties?

Maybe add some of these details to the comments?

Reply:
Using some of the arguments from [`s3_bucket.html.markdown`](https://raw.githubusercontent.com/terraform-providers/terraform-provider-aws/master/website/docs/r/s3_bucket.html.markdown) as an example, I want to generate a map like this:

- bucket
    - description: "(Optional, Forces new resource) The name of the bucket. If omitted, Terraform will assign a random, unique name."
    - arguments: nil
- website
    - description: "(Optional) A website object (documented below)."
    - arguments:
        - index_document: "(Required, unless using `redirect_all_requests_to`) Amazon S3 returns this index document when requests are made to the root domain or any of the subfolders."
- index_document
    - description: "(Required, unless using `redirect_all_requests_to`) Amazon S3 returns this index document when requests are made to the root domain or any of the subfolders."
    - arguments: nil
   - isNested: true

where both `bucket` and `website` are top-level properties, but `website` is also an object with its own arguments. I'm *guessing* there's not many instances of repeated fields with the same parent name [1], so I only go a single nest deep.

I'll add a comment to (hopefully) clear this up a bit more.


[1] For example, suppose `website` has a `website` field that also has a `website` field; then `map["website"].arguments["website"]` could actually refer to both our top-level `website`'s `website` argument or that `website`'s `website` argument.